R/tmb.R

```diff
@@ -427,6 +427,25 @@ h_mmrm_tmb_fit <- function(tmb_object,
   )
 }
 
+h_get_x_matrix <- function(object, newdata) {
+  # TODO: double check that this is even necessary -> emmeans integration
+  # get model frame used for fitting and extract factor levels
+  full_frame <- object$tmb_data$full_frame
+  xlevels <- list()
+  for (i in seq_along(full_frame)) {
+    name <- names(full_frame)[i]
+    if (is.factor(full_frame[[i]]) & name %in% names(newdata)) {
+      xlevels[[name]] <- levels(full_frame[[i]])
+    }
+  }
+  # get terms object without response variable
+  fterms <- stats::delete.response(stats::terms(object$formula_parts$model_formula))
+  # first construct model frame, then the x_matrix
+  mframe <- model.frame(fterms, data = newdata)
```

Comment:
`model.frame.mmrm_tmb` already implemented and supports new data now. I think we can use that (or update it as needed)

Reply:
@clarkliming could you squeeze that in here? I find the structure a bit confusing and don't want to make too many assumptions. We can keep the helper function for testing and if we reduce it to 2-3 lines with model.frame.mmrm_tmb we can remove it again.

Reply:
see #262

Reply:
Hm ... actually @yonicd just mentioned to me rightfully that `model.matrix()` below won't include the response anyway... so we don't need to worry about dropping the response at all.